ieeeuoft/hackathon-template
# Leave Team API View

## Overview

- Resolves IEEE-37
- Add API endpoint for leaving event team
- The request is served in an atomic action to prevent the case where another user joins the team while the team is getting deleted, which would cascade the delete
- Endpoint can only be accessed if user is signed in and already on a team
- User can leave team if there is no not-cancelled orders on the team
- User will be put on a new team
- If old team is now empty, delete the old team


## Unit Tests Created

- Test access
- Test leaving and creating team
- Test can only leave when orders are cancelled
- Test old team deleted if empty


## Steps to QA

- Make a profile for your user, call the endpoint
  - Should return a serialized team object for a new team
  - db should not have any other team
- Add another user to your team, call the endpoint
  - Should return a serialized team object for a new team
  - db should have both teams
- Add some orders to your team
  - Try different values for the status of the order
  - Call the endpoint
    - The request should only succeed when the order is cancelled
      - Check that the old team is no longer in the db, and that the orders are deleted as well
    - Otherwise, the request should return a 400
